python/tests/test_combinatorics.py

```diff
@@ -420,3 +433,469 @@ def test_is_symmetrical(self):
+        0.000000    1.000000    7  5
+        0.000000    1.000000    7  6
+        0.000000    1.000000    8  4
+        0.000000    1.000000    8  7
+        """
+        )
+        ts = tskit.load_text(
+            nodes, edges, sequence_length=1, strict=False, base64_metadata=False
+        )
+
+        sample_sets = [[0], [2], [4]]
+
+        expected = comb.TopologyCounter()
+        expected[0] = collections.Counter({(0, 0): 1})
+        expected[1] = collections.Counter({(0, 0): 1})
+        expected[2] = collections.Counter({(0, 0): 1})
+        expected[0, 1] = collections.Counter({(0, 0): 1})
+        expected[0, 2] = collections.Counter({(0, 0): 1})
+        expected[1, 2] = collections.Counter({(0, 0): 1})
+        expected[0, 1, 2] = collections.Counter({(1, 2): 1})
+
+        tree_topologies = ts.first().count_topologies(sample_sets)
+        treeseq_topologies = list(ts.count_topologies(sample_sets))
+        self.assertEqual(tree_topologies, expected)
+        self.assertEqual(treeseq_topologies, [expected])
+
+    def test_internal_samples_errors(self):
+        tables = tskit.TableCollection(sequence_length=1.0)
+
+        c1 = tables.nodes.add_row(flags=tskit.NODE_IS_SAMPLE, time=0)
+        c2 = tables.nodes.add_row(flags=tskit.NODE_IS_SAMPLE, time=0)
+        p = tables.nodes.add_row(flags=tskit.NODE_IS_SAMPLE, time=1)
+
+        tables.edges.add_row(left=0, right=1, parent=p, child=c1)
+        tables.edges.add_row(left=0, right=1, parent=p, child=c2)
+
+        self.verify_value_error(tables.tree_sequence())
+
+    def test_non_sample_nodes_errors(self):
+        tables = tskit.TableCollection(sequence_length=1.0)
+
+        c1 = tables.nodes.add_row(flags=tskit.NODE_IS_SAMPLE, time=0)
+        c2 = tables.nodes.add_row(time=0)
+        p = tables.nodes.add_row(time=1)
+
+        tables.edges.add_row(left=0, right=1, parent=p, child=c1)
+        tables.edges.add_row(left=0, right=1, parent=p, child=c2)
+
+        sample_sets = [[0], [1]]
+        self.verify_value_error(tables.tree_sequence(), sample_sets)
+
+        sample_sets = [[0], [-1]]
+        self.verify_node_out_of_bounds_error(tables.tree_sequence(), sample_sets)
+
+    def verify_value_error(self, ts, sample_sets=None):
+        with self.assertRaises(ValueError):
+            ts.first().count_topologies(sample_sets)
+        with self.assertRaises(ValueError):
+            list(ts.count_topologies(sample_sets))
+
+    def verify_node_out_of_bounds_error(self, ts, sample_sets=None):
+        with self.assertRaises(ValueError):
+            ts.first().count_topologies(sample_sets)
+        with self.assertRaises(IndexError):
+            list(ts.count_topologies(sample_sets))
+
+    def test_standard_msprime_migrations(self):
+        for num_populations in range(2, 5):
+            samples = [5] * num_populations
+            ts = self.simulate_multiple_populations(samples)
+            self.verify_topologies(ts)
+
+    def test_complex_dtwf_topologies(self):
```

Comment:
This is good, but I was thinking more about the *forwards time* trees that we get out of the local Wright-Fisher simulator. Something like [these three tests](https://github.com/tskit-dev/tskit/blob/master/python/tests/test_stats.py#L293) would be really good. We want to see, in particular, how it deals with the nasty raggedy trees that a WF simulation produces.

Reply:
Those tests work great just drop in. Cool also to test against all sample set partitions!